scripts/on_sd_start.sh

```diff
@@ -63,11 +63,30 @@ case "${OS_NAME}" in
     *)          echo "Unknown OS: $OS_NAME! This script runs only on Linux or Mac" && exit
 esac
 
+# Detect GPU types
+
+if grep -q amdgpu /proc/bus/pci/devices; then
+   echo AMD GPU detected
+   HAS_AMD=yes
+fi
+
+if grep -q nvidia /proc/bus/pci/devices; then
+   echo NVidia GPU detected
+   HAS_NVIDIA=yes
+fi
+
+
+
 # install torch and torchvision
 if python ../scripts/check_modules.py torch torchvision; then
     # temp fix for installations that installed torch 2.0 by mistake
     if [ "$OS_NAME" == "linux" ]; then
-        python -m pip install --upgrade torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url https://download.pytorch.org/whl/cu116 -q
+        # Check for AMD and NVIDIA dGPUs, always preferring an NVIDIA GPU if available
+        if [ "$HAS_NVIDIA" != "yes" -a "$HAS_AMD" = "yes" ]; then
+            python -m pip install --upgrade torch torchvision --extra-index-url "https://download.pytorch.org/whl/rocm5.4.2"  || fail "Installation of torch and torchvision for AMD failed"
+        else
+            python -m pip install --upgrade torch==1.13.1+cu116 torchvision==0.14.1+cu116 --extra-index-url "https://download.pytorch.org/whl/cu116"  || fail "Installation of torch and torchvision for CUDA failed"
```

Comment:
Can you please add back the `-q` quiet flag? These lines are run each time the program starts up, to force the package back to the specified version of torch.

Reply:
Please don't run stuff quietly. Makes it really hard if you have to debug stuff that you can't see.